### PR TITLE
Translation crud

### PR DESCRIPTION
**Pull request by BaDo2001**

### CODE REVIEW

File: admin-site/app/api/(routes)/projects/keys/route.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { type NextRequest, NextResponse } from "next/server";
+import * as z from "zod";
+
+import { requireProjectApiKey } from "@/api/utils/requireProjectApiKey";
+
+import { saveNewKey } from "../../../keys/addKey";
```
Comment (tothpeterdev) [inline admin-site/app/api/(routes)/projects/keys/route.ts:6]: We should use the @/api path alias

File: admin-site/app/api/types/project.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import type { Member, Project, Translation } from "@prisma/client";
+
+export type ProjectWithTranslations = Project & { translations: Translation[] };
```
Comment (tothpeterdev) [inline admin-site/app/api/types/project.ts:3]:
```suggestion
export type ProjectWithTranslations = ProjectIncludes<false, true>;
```

File: admin-site/app/api/keys/addKey.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    project,
+    key,
+    value,
+  });
+};
+
+type SaveNewKeyArgs = Omit<AddKeyArgs, "projectId"> & {
+  project: Project & {
+    translations: Translation[];
+  };
```
Comment (tothpeterdev) [inline admin-site/app/api/keys/addKey.ts:32]:
```suggestion
  project: ProjectWithTranslations;
```

**Comment by aront22** [comment]
Cannot access the from Postman:
INFO: Clerk: The request to /api/projects/translations is being protected (401) because there is no signed-in user, and the path is included in `apiRoutes`. To prevent this behavior, choose one of:

1. To prevent Clerk authentication from protecting (401) the api route, remove the rule matching "/api/projects/translations" from the `apiRoutes` array passed to authMiddleware
2. To make the route accessible to both signed in and signed out users, add "/api/projects/translations" to the `publicRoutes` array passed to authMiddleware     
3. To prevent Clerk authentication from running at all, pass `ignoredRoutes: ["/((?!api|trpc))(_next.*|.+\.[\w]+$)", "/api/projects/translations"]` to authMiddleware
4. Pass a custom `afterAuth` to authMiddleware, and replace Clerk's default behavior of redirecting unless a route is included in publicRoutes

For additional information about middleware, please visit https://clerk.com/docs/nextjs/middleware
(This log only appears in development mode, or if `debug: true` is passed to authMiddleware)

We should use use custom authentication in this case by checking the api key passed in the Authorization header.

File: admin-site/app/api/(routes)/projects/keys/route.ts (hunk 1 above)
Comment (BaDo2001) [reply admin-site/app/api/(routes)/projects/keys/route.ts:6]: Done

File: admin-site/app/api/types/project.ts (hunk 2 above)
Comment (BaDo2001) [reply admin-site/app/api/types/project.ts:3]: Done

File: admin-site/app/api/keys/addKey.ts (hunk 3 above)
Comment (BaDo2001) [reply admin-site/app/api/keys/addKey.ts:32]: Done

File: admin-site/utils/formatTranslations.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import type { Translation, TranslationEntry } from "@prisma/client";
+import _ from "lodash";
+
+import { createNestedObject } from "./createNestedObject";
+
+type TranslationWithEntries = Translation & {
```
Comment (tothpeterdev) [inline admin-site/utils/formatTranslations.ts:6]: We could introduce a TranslationIncludes type similar to ProjectIncludes, then use it to define the TranslationWithEntries type

File: admin-site/prisma/schema.prisma [hunk 5]
```diff
@@ -13,7 +13,7 @@ model Project {
   name            String  @unique
   ownerId         String
   defaultLanguage String
-  apiKey          String?
+  apiKey          String? @unique
 
   translations Translation[] @relation()
```
Comment (tothpeterdev) [inline admin-site/prisma/schema.prisma:18]:
```suggestion
  translations Translation[]
```

File: admin-site/utils/formatTranslations.ts (hunk 4 above)
Comment (BaDo2001) [reply admin-site/utils/formatTranslations.ts:6]: This one is only used here so far and also translation only has one relation whereas project has more.

File: admin-site/prisma/schema.prisma (hunk 5 above)
Comment (BaDo2001) [reply admin-site/prisma/schema.prisma]: Fixed

**Review by tothpeterdev — APPROVED** [review]

**Pull request merged**